File: LeetCode-2023/2042.检查句子中的数字是否递增.py
#
# @lc app=leetcode.cn id=2042 lang=python3
#
# [2042] 检查句子中的数字是否递增
#

# @lc code=start
from curses.ascii import isdigit
from posixpath import split


class Solution:
    def areNumbersAscending(self, s: str) -> bool:
        ss = [int(i) for i in s.split() if i.isdigit()]
        for i in range(len(ss) - 1):
            if (ss[i] >= ss[i+1]):
                return False
        return True

# @lc code=end

File: LeetCode-2023/2042.检查句子中的数字是否递增.cpp
/*
 * @lc app=leetcode.cn id=2042 lang=cpp
 *
 * [2042] 检查句子中的数字是否递增
 */

// @lc code=start
class Solution {
public:
    bool areNumbersAscending(string s) {
        stringstream ss(s);
        string token;
        int baseNumber = 0;
        while(ss >> token) {
            try
            {
                int number = stoi(token);
                if(number > baseNumber) {
                    baseNumber = number;
                } else {
                    return false;
                }
            }
            catch(const std::exception& e)
            {
                std::cerr << e.what() << '\n';
            }
        }
        return true;
    }
};
// @lc code=end

